packages/client-node/src/result_set.ts

```diff
@@ -82,16 +82,12 @@ export class ResultSet<Format extends DataFormat | unknown>
     // JSONEachRow, etc.
     if (isStreamableJSONFamily(this.format as DataFormat)) {
       const result: T[] = []
-      await new Promise((resolve, reject) => {
-        const stream = this.stream<T>()
-        stream.on('data', (rows: Row[]) => {
-          for (const row of rows) {
-            result.push(row.json())
-          }
-        })
-        stream.on('end', resolve)
-        stream.on('error', reject)
```

Comment:
not needed?

Reply:
This is the exact part that produced unhandled exceptions.

```ts
for await (const rows of stream) {
  for (const row of rows) {
    result.push(row.json())
  }
}
```
achieves the same on the modern Node.js versions, but we can catch the error properly o the app side.